script/dev/Swap.s.sol

```diff
@@ -102,31 +101,62 @@ contract Swap is PredicateHelpers {
         bool zeroForOne,
         bool withPredicateMessage
     ) internal {
+        bytes memory actions;
         bytes[] memory inputs = new bytes[](1);
-        bytes memory actions = abi.encodePacked(
-            uint8(Actions.SWAP_EXACT_OUT_SINGLE),
-            uint8(Actions.TAKE_ALL),
-            uint8(Actions.SETTLE_ALL)
-        );
-
-        uint128 swapAmountOut = uint128(swapAmount);
-        uint128 swapAmountIn = type(uint128).max;
-
         bytes[] memory swapParams = new bytes[](3);
-        swapParams[0] = abi.encode(
-            IV4Router.ExactOutputSingleParams({
-                poolKey: poolKey,
-                zeroForOne: zeroForOne,
-                amountOut: swapAmountOut,
-                amountInMaximum: swapAmountIn,
-                hookData: withPredicateMessage
-                    ? abi.encode(_getPredicateMessage(caller, poolKey, hook, zeroForOne, int256(swapAmount)))
-                    : abi.encode("")
-            })
-        );
 
-        swapParams[1] = abi.encode(poolKey.currency0, swapAmountOut);
-        swapParams[2] = abi.encode(poolKey.currency1, swapAmountIn);
+        uint128 swapAmountOut;
+        uint128 swapAmountIn;
+
+        if (zeroForOne) {
+            actions = abi.encodePacked(
+                uint8(Actions.SWAP_EXACT_IN_SINGLE),
+                uint8(Actions.SETTLE_ALL),
+                uint8(Actions.TAKE_ALL)
+            );
+
+            swapAmountOut = 0;
```

Comment:
no slippage protection, suggestion to add add configurable parameter with 0.01% slippage allowed by default

Reply:
Added in this commit: https://github.com/m0-foundation/uniswap-v4-hooks/pull/20/commits/c7d4a7a373ef27c8d139547ea0dbc7f363a70c8a
Can be set to 0 for the wM/USDC pool once we will run the scripts again for these pools.